src/dosext/builtins/builtins.c

```diff
@@ -320,17 +320,32 @@ int com_dossetcurrentdir(char *path)
 
 static int is_valid_drive(int drv)
 {
-  uint16_t ret;
+  char *fname, *fcb;
+  int ret;
 
   pre_msdos();
 
-  HI(ax) = 0x36; // get free disk space
-  LO(dx) = drv;
+  /* Parse filename into FCB (physical, formatted or not, and network) */
+  fname = lowmem_heap_alloc(16);
+  snprintf(fname, 16, "%c:FILENAME.EXT", 'A' - 1 + drv);
```

Comment:
Maybe use "NUL" here?

Reply:
Could do but it's not testing for existence, just that the name is parseable and the drive is valid.

Reply:
Is there even no need for `\` after `:`?

Reply:
Actually it won't work if you do that, it's a DOS v1 function and that had no directories